zerver/data_import/hipchat.py

```diff
@@ -439,19 +441,38 @@ def get_raw_messages(fn: str) -> List[ZerverFieldsT]:
             if message_key in d
         ]
 
-        return [
-            dict(
+        def get_raw_message(d: Dict[str, Any]) -> Optional[ZerverFieldsT]:
+            if isinstance(d['sender'], str):
+                # Some Hipchat instances just give us a person's
+                # name in the sender field for NotificationMessage.
+                # We turn them into a mirror user.
+                mirror_user = user_handler.get_mirror_user(
+                    realm_id=realm_id,
+                    name=d['sender'],
+                )
+                sender_id = mirror_user['id']
+            else:
+                sender_id = d['sender']['id']
+
+            return dict(
                 fn_id=fn_id,
-                sender_id=d['sender']['id'],
+                sender_id=sender_id,
                 receiver_id=d.get('receiver', {}).get('id'),
                 content=d['message'],
-                mention_user_ids=d['mentions'],
+                mention_user_ids=d.get('mentions', []),
                 pub_date=str_date_to_float(d['timestamp']),
-                attachment=d['attachment'],
+                attachment=d.get('attachment'),
                 files_dir=files_dir,
             )
-            for d in flat_data
-        ]
+
+        raw_messages = []
+
+        for d in flat_data:
+            raw_message = get_raw_message(d)
+            if raw_message is not None:
```

Comment:
I don't see under what circumstance `get_raw_message` would ever return `None`?

Reply:
That's an artifact of when I was punting on NotificationMessage records due to the mirror-user issue.  I can clean that up.